### PR TITLE
Explain why awaitRunningTasks leaves one active

### PR DESCRIPTION
**Pull request by mihails-strasuns-sociomantic**
Per slack discussion

### CODE REVIEW

**Comment by mihails-strasuns-sociomantic** [comment]
FYI @daniel-zullo-sociomantic

File: src/ocean/task/TaskPool.d [hunk 1]
```diff
@@ -160,14 +160,23 @@ class TaskPool ( TaskT : Task ) : ObjectPool!(Task)
         Suspends the current task until all running tasks in the pool have
         finished executing.
 
+        Because of `terminationHook` implementation details, by the time caller
+        task gets resumed, there will still be one (last) non-recyled running
+        task in the pool, suspended right before actual termination. Once caller
+        tasks gets suspended again for any reason, that last task will be
```
Comment (daniel-zullo) [inline src/ocean/task/TaskPool.d]: I think it should be `Once caller task` here (task without `s` at the end)

File: src/ocean/task/TaskPool.d [hunk 2]
```diff
@@ -160,14 +160,23 @@ class TaskPool ( TaskT : Task ) : ObjectPool!(Task)
...
+        recyled too. It is possible to manually call
```
Comment (daniel-zullo) [inline src/ocean/task/TaskPool.d]: typo `recyled`

**Review by daniel-zullo — COMMENTED** [review]
2 cosmetic comments and LGTM.

File: src/ocean/task/TaskPool.d [hunk 3]
```diff
@@ -160,14 +160,23 @@ class TaskPool ( TaskT : Task ) : ObjectPool!(Task)
         Suspends the current task until all running tasks in the pool have
         finished executing.
 
+        Because of `terminationHook` implementation details, by the time caller
+        task gets resumed, there will still be one (last) non-recyled running
```
Comment (daniel-zullo) [inline src/ocean/task/TaskPool.d]: typo also here in `non-recyled`

File: src/ocean/task/TaskPool.d [hunk 4]
```diff
@@ -160,14 +160,23 @@ class TaskPool ( TaskT : Task ) : ObjectPool!(Task)
...
+        task in the pool, suspended right before actual termination. Once caller
+        tasks gets suspended again for any reason, that last task will be
+        recyled too. It is possible to manually call
+        `theScheduler.processEvents()` after `awaitRunningTasks()` to force
+        recycling of that last task at cost of small additional delay in
```
Comment (daniel-zullo) [inline src/ocean/task/TaskPool.d]: Maybe prepend **a** to `small addition delay...`

**Comment by mihails-strasuns-sociomantic** [comment]
- fixed grammar

**Pull request merged**